dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/ReturnDatasetToAuthorCommand.java

```diff
@@ -52,12 +53,16 @@ public Dataset execute(CommandContext ctxt)  {
             Then remove reviewers from the autors list
             Finally send a notification to the remaining (non-reviewing) authors - Hey! your dataset was rejected.
         */
+        AuthenticatedUser requestor = getUser().isAuthenticated() ? (AuthenticatedUser) getUser() : null;
         List<AuthenticatedUser> reviewers = ctxt.permissions().getUsersWithPermissionOn(Permission.PublishDataset, savedDataset);
         List<AuthenticatedUser> authors = ctxt.permissions().getUsersWithPermissionOn(Permission.EditDataset, savedDataset);
         authors.removeAll(reviewers);
         for (AuthenticatedUser au : authors) {
-            ctxt.notifications().sendNotificationWithEmail(au, getTimestamp(), NotificationType.RETURNEDDS,
-                                                           savedDataset.getLatestVersion().getId(), NotificationObjectType.DATASET_VERSION, comment);
+            Option.of(requestor)
+                    .peek(user -> ctxt.notifications().sendNotificationWithEmail(au, getTimestamp(), NotificationType.RETURNEDDS,
+                            savedDataset.getLatestVersion().getId(), NotificationObjectType.DATASET_VERSION, requestor, comment))
+                    .onEmpty(() -> ctxt.notifications().sendNotificationWithEmail(au, getTimestamp(), NotificationType.RETURNEDDS,
```

Comment:
Can user who is not authenticated send comment?

Reply:
This I've to be consistent with `SubmitDatasetForReviewCommand`. I guess asking @wfenrich might clarify this.